riscv/src/small_field/bootloader.rs

```diff
@@ -1 +1,100 @@
+use std::marker::PhantomData;
 
+use powdr_number::KnownField;
+use powdr_number::{FieldElement, LargeInt};
+
+use crate::continuations::bootloader::BootloaderImpl;
+
+use crate::continuations::bootloader::WORDS_PER_PAGE;
+
+pub fn split_word<F: FieldElement>(v: u32) -> (F, F) {
+    ((v & 0xffff).into(), (v >> 16).into())
+}
+
+pub fn split_fe<F: FieldElement>(v: &F) -> [F; 2] {
+    let v = v.to_integer().try_into_u32().unwrap();
+    [(v >> 16).into(), (v & 0xffff).into()]
+}
+
+pub struct SmallFieldBootloader<F: FieldElement>(PhantomData<F>);
+
+impl<F: FieldElement> BootloaderImpl<F> for SmallFieldBootloader<F> {
+    const FE_PER_WORD: usize = 2;
+    type Page = [F; 2 * WORDS_PER_PAGE];
+    type Hash = [F; 8];
+
+    fn update_page(page: &mut Self::Page, idx: usize, word: u32) {
+        let (hi, lo) = split_word(word);
+        // TODO: check proper endianess here!
```

Comment:
I think RISCV memory is [low, high], so this should fail

Reply:
yes, riscv is little endian, but i was not sure here (we use hi,lo in our interface to the memory machine), so I left the todo

Reply:
actually yea, it's tricky. We use big-endian in the machine, but Rust uses little endian when representing u64 as 2 words in RISCV32, for example. So this is probably correct actually